src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/VariadicValueResolver.php

```diff
@@ -41,8 +41,6 @@ public function resolve(Request $request, ArgumentMetadata $argument)
             throw new \InvalidArgumentException(sprintf('The action argument "...$%1$s" is required to be an array, the request attribute "%1$s" contains a type of "%2$s" instead.', $argument->getName(), \gettype($values)));
         }
 
-        foreach ($values as $value) {
-            yield $value;
-        }
+        yield from $values;
```

Comment:
@iltar: is the change safe here?

Reply:
@kalessil What the code here does is transforming `&arr[]=foo&arr[]=bar` to `['foo', 'bar']` in php, which then get yielded to provide support for `action(...$arr)` I believe. I think @stof made a valid remark and I don't know if this will break anything, as variadic arguments can only receive numeric keys afaik. Would require the addition of a test case to verify if non-numeric keys will work here (which I fear do not).

Reply:
the question is whether the value resolver will keep working fine if `$value` is an associative array (which is totally possible, as this is something coming from outside Symfony itself)

Reply:
Right, `action(...$arr)` requires array to be indexed from 0 to N sequentially. So, the change will be BC if someone provides indexes. `yield from array_values($values);` would be a possible solution.

Reply:
but that force to first allocate a new array to iterate over it, in which case the existing code is better (it avoids the allocation of the array)

Reply:
Agree, let's revert the both cases you've mentioned. I obviously need to differentiate case with single value and key-value pair yielding in my plugin.